### ISSUE TITLE
Support compressing files bigger than 1MB?

### ISSUE THREAD

**Issue by CarterLi** [RomanBelkov/XZ.NET#2]
1. Uncomment "Simple compression (with timing)" part in Examples.cs; rebuild
2. Copy & paste a big file into "XZ.NET\Examples\bin\Debug", rename it to "test.img"
3. Run Examples.exe

Result: test.img.xz is generated, but only 1MB. Try to uncompress it using 7Zip, get Data error

![image](https://cloud.githubusercontent.com/assets/6134068/7553692/2cef5784-f739-11e4-83f2-ab58354e18f8.png)

**Comment by RomanBelkov** [comment]
I guess that this bug is true for x86-64 version of XZ.NET that You have introduced in #1, because x86 version works just fine (for example, it is used in [DiskImager](https://github.com/RomanBelkov/DiskImager)).

I'l look into the matter and will give an answer ASAP.

**Comment by CarterLi** [comment]
I found this when testing my x86-64 version. But of course, I have tested this on the original version too.
![image](https://cloud.githubusercontent.com/assets/6134068/7554015/ca6e1b96-f746-11e4-9a8c-34cee201f4ae.png)

VS2013 Update4 on Win7 x64
I just uncommented the code in Project Examples, nothing else.

EDIT: Compressing small files ( smaller than 1MB ) works on my x64 verson too as it does on the original version.

**Comment by RomanBelkov** [comment]
Strange, but x86-64 works fine with my VS2013 on Win 8.1 Pro

Used a 1.5 Gb .img firmware for a robot

![proof1](https://cloud.githubusercontent.com/assets/5811999/7554144/27cfa7e0-f725-11e4-8f43-907e3a122d57.PNG)
![proof2](https://cloud.githubusercontent.com/assets/5811999/7554145/27d2bc64-f725-11e4-9e72-3049d977032a.PNG)

**Comment by CarterLi** [comment]
Weird... May caused by the installation of VS2015 RC on my mathine?
Please keep this open, I'll test it again when VS2015 is RTMed

**Comment by RomanBelkov** [comment]
OK, I'll keep this open and also will test the matter on some other machines.

**Comment by RomanBelkov** [comment]
CarterLi, can You please provide Your e-mail as I'd like to discuss one moment

**Comment by CarterLi** [comment]
512147886@qq.com
But why not discuss it here?

**Comment by RomanBelkov** [comment]
I wanted to discuss switching the license to MIT and it would be a bit off the theme of this issue.

Speaking on the matter: If I have a chance, I will test the library on VS2015 in a couple of days

**Comment by CarterLi** [comment]
> I wanted to discuss switching the license to MIT

It's ok

**Comment by RomanBelkov** [comment]
So...

I tried running an example again after recent Windows re-install and surprisingly it failed. At this point I am a bit clueless but I am almost sure that something is broken in the example. I will look into the matter ASAP.

Currently the library is working just fine in this program:
https://github.com/RomanBelkov/DiskImager/blob/master/DiskImager/Disk.cs

**Comment by philxyz** [comment]
This looks similar to what I was seeing before I sent the last patch. Does it work correctly now?

**Comment by PhilAtWysdom** [comment]
Is this working now in master? I noticed a similar issue with filesize before applying that last change.

**Comment by RomanBelkov** [comment]
Sorry, I'm busy right now and will answer ASAP (probably in 2 days or so)

**Comment by RomanBelkov** [comment]
Nope, that does not solve an issue for me. I'll look into the matter and give a reply as soon as I found anything

**Comment by RomanBelkov** [comment]
Everything seems to be OK right now. Please check and comment

**Comment by PhilAtWysdom** [comment]
I haven't personally experienced the problem described in this ticket - but I did have another issue, which my patch addressed. I was just curious as to whether it also cured this ticket since it was the only one. If it works for you (for your test case) where it didn't before, that sounds good to me. I have retested my use case also and the library is still working great here. Thanks for your attention and support
